stix_shifter_modules/guardium/stix_transmission/guard_utils.py

```diff
@@ -1,25 +1,29 @@
-import requests
-from requests.models import Response
-import json
-import sys, argparse, traceback
+import argparse
+import datetime
 import hashlib
-import datetime, re
+import json
+import re
+import sys
+import traceback
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
 from stix_shifter_utils.utils import logger
 
-class GuardApiClient(object):
 
-    def __init__ (self,client_id, url, secret, user, password):
-        super().__init__()
+class GuardApiClient(RestApiClient):
+
+    def __init__(self, params, host, port, headers,
+                 url_modifier_function, cert_verify, sni, auth):
+        self.client = RestApiClient(host, port, headers, url_modifier_function, cert_verify, sni, auth)
```

Comment:
I am not quite sure about this. GuardApiClient inherits from RestApiClient and has a property (self.client) that is again RestApiClient...
Is this solution about to use self.client to get token?

Reply:
Yes. It uses `self.client` for all of the requests. First - ApiClient and GuardApiClient have a client field and it's legal to have an attribute with the same name in two independent classes. Second - calling `self.client = RestApiClient(host, port, headers, url_modifier_function, cert_verify, sni, auth)` is the same as calling `self.client =  super().__init__(host, port, headers, url_modifier_function, cert_verify, sni, auth)` which is a must in a child class.

Would it be better if I change the name of ApiClient's client field to `aux_client`, or something of the sort, for better readability?

Reply:
I guess keeping it as just client is also good in this case